molecule/drupal/converge.yml

```diff
@@ -0,0 +1,244 @@
+---
+- name: Converge
+  hosts: instance
+  become: true
+  pre_tasks:
+    - name: Add ppa:ondrej/php repository
+      apt_repository:
+        repo: ppa:ondrej/php
+      when: ansible_os_family == 'Debian'
+    - name: Import remi GPG key.
+      rpm_key:
+        key: "https://rpms.remirepo.net/RPM-GPG-KEY-remi2018"
+        state: present
+      when: ansible_os_family == 'RedHat'
+    - name: Add remi-php74 repository
+      yum:
+        name: https://rpms.remirepo.net/enterprise/remi-release-{{ ansible_distribution_major_version }}.rpm
+        state: present
+      when: ansible_os_family == 'RedHat'
+    - name: Enable module php7.4 for Centos8
+      dnf:
+        name: '@php:remi-7.4'
+        state: present
+      when: ansible_os_family == 'RedHat' and ansible_distribution_major_version == '8'
+    - name: Install python3 Psycopg2
+      apt:
+        name: python3-psycopg2
+      when: ansible_os_family == 'Debian'
+    - name: "Set ansible_python_interpreter for Ubuntu 20"
+      set_fact:
+        ansible_python_interpreter: "/usr/bin/python3"
+      when: ansible_distribution == 'Ubuntu' and ansible_distribution_major_version | int >= 18
+    - name: "Set nginx extra_parameters"
+      set_fact:
+        nginx_extra_parameters: |
+          client_max_body_size 4096M;
+
+          location / {
+              try_files $uri /index.php?$query_string;
+          }
+
+          location @rewrite {
+            rewrite ^ /index.php;
+          }
+
+          location ~ '\.php$|^/update.php' {
+            fastcgi_split_path_info ^(.+?\.php)(|/.*)$;
+            try_files $fastcgi_script_name =404;
+            include fastcgi_params;
+            fastcgi_param HTTP_PROXY "";
+            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+            fastcgi_param PATH_INFO $fastcgi_path_info;
+            fastcgi_param QUERY_STRING $query_string;
+            fastcgi_intercept_errors on;
+            fastcgi_pass 127.0.0.1:9000;
+          }
+
+          location ~* \.(js|css|png|jpg|jpeg|gif|ico|svg)$ {
+            try_files $uri @rewrite;
+            expires max;
+            log_not_found off;
+          }
+
+          location ~ ^/sites/.*/files/styles/ {
+            try_files $uri @rewrite;
+          }
+
+          location ~ ^(/[a-z\-]+)?/system/files/ {
+            try_files $uri /index.php?$query_string;
+          }
+
+          if ($request_uri ~* "^(.*/)index\.php/(.*)") {
+            return 307 $1$2;
+          }
+
+          error_log /var/log/nginx/project_error.log;
+          access_log /var/log/nginx/project_access.log;
+    - name: "Set steamengine_deployments var"
+      set_fact:
+        steamengine_deployments:
```

Comment:
As I said, you could share some stuff between scenarios. testinfra tests are duplicated 4 times. 80% of the converge playbook is duplicated. requirements.yml is duplicated.

All of this can be placed in a directory "drupal-shared" and included in each scenario (using the right instruction in molecule.yml).